src/controllers/admin.category.controller.ts

```diff
@@ -42,3 +47,41 @@ export const createCategory = [
     return;
   }),
 ];
+
+const checkExistsCategory = async (req: IAdminCategoryRequest, res: Response, next: NextFunction) => {
+  const category = await categoryService.getCategoryById(parseInt(req.params.id));
+  if (category === null) {
+    return res.render('error', { code: ErrorCode.NOT_FOUND, title: t('error.notFound'), message: t('error.notFound', {id: req.params.id}) });
+  }
+
+  req.category = category;
+  next();
+};
+
+
+export const updateCategory = [
+  checkLoggedIn,
+  checkIsAdmin,
+  checkValidId,
+  checkExistsCategory,
+  validateCreateCategory,
+  asyncHandler(async (req: IAdminCategoryRequest, res: Response) => {    
+    const category = await categoryService.getCategoryByName(req.body.name);
+
+    if (category) {
+      res.send({
+        status: Status.FAIL,
+        message: getTranslatedMessage('error.categoryExist', req.query.lng),
+      });
+      return;
+    }
```

Comment:
phần check category bị trùng tên có xử lý cả ở action create và update -> e tách thành function not export để tránh lặp code nhé